### DISCUSSION TITLE
[Feature Request] Option to fast forward voice notes, like in WhatsApp

### DISCUSSION THREAD

**Discussion by NMBlanco** [rodit/SnapMod#100] [Ideas & Feature Suggestions]
Like 1,5×, 2× speed. Maybe it's possible. Appreciate your constant updates btw! 🔥

**Comment by Magna-Graecia** [comment]
Close this and open it in the 'Discussions' tab, bro. Rodit recently made an announcement: https://github.com/rodit/SnapMod/discussions/92

**Comment by rodit** [comment]
Added in 1.5.8 - you can set the global playback speed in SnapMod settings (which applies to all audio notes). You can also hold down on an audio note and press 'Set Playback Speed' in the context menu. This will override the global playback speed and the playback speed is retained for that audio note until Snapchat is closed (killed).

**Reply by NMBlanco** [reply]
> Added in 1.5.8 - you can set the global playback speed in SnapMod settings (which applies to all audio notes). You can also hold down on an audio note and press 'Set Playback Speed' in the context menu. This will override the global playback speed and the playback speed is retained for that audio note until Snapchat is closed (killed).

Wow that was fast! Appreciate it a lot <3

**Reply by Magna-Graecia** [reply]
Yes indeed it was! I'm happy to report that everything is working well on my end on the latest version, including the new voice note playback feature, as well as chat snap saves that was previously broken. Thanks @rodit !

**Reply by Magna-Graecia** [reply]
@rodit By the way, one thing I noticed is that audio notes now look different after updating to Snapchat 11.78.1.39 and SnapMod v1.6.2. Instead of just being a circle like it was previously, it's rectangular and shows the volume graph or whatever you call it within the rectangle. It also has a speed toggle on the right side of the note. Default is 1x. Tapping on it once changes it to 1.5x. Tapping again changes it to 2x. And tapping once more cycles back to 1x. Is this a new Snapchat feature for 11.78.1.39? Or is it a SnapMod modification?

Another thing I noticed is that holding down on the note no longer brings up the "Set Playback Speed" option, so I'm thinking this might all be your doing haha. Was 2x the max speed that you could previously set the playback speed to? Because of course now it is since it toggles back to 1x after 2x, and there's no more "Set Playback Speed" option where you manually enter the number. I personally never entered higher than 2.0 in that box when testing, so I wouldn't know haha. Maybe 2.0 was the max speed and you just streamlined it all.

**Reply by NMBlanco** [reply]
> @rodit By the way, one thing I noticed is that audio notes now look different after updating to Snapchat 11.78.1.39 and SnapMod v1.6.2. Instead of just being a circle like it was previously, it's rectangular and shows the volume graph or whatever you call it within the rectangle. It also has a speed toggle on the right side of the note. Default is 1x. Tapping on it once changes it to 1.5x. Tapping again changes it to 2x. And tapping once more cycles back to 1x. Is this a new Snapchat feature for 11.78.1.39? Or is it a SnapMod modification?
> 
> Another thing I noticed is that holding down on the note no longer brings up the "Set Playback Speed" option, so I'm thinking this might all be your doing haha. Was 2x the max speed that you could previously set the playback speed to? Because of course now it is since it toggles back to 1x after 2x, and there's no more "Set Playback Speed" option where you manually enter the number. I personally never entered higher than 2.0 in that box when testing, so I wouldn't know haha. Maybe 2.0 was the max speed and you just streamlined it all.

Weird, audio notes still look the same for me on that version. Maybe it's some server sided test for random selected users?

**Reply by Magna-Graecia** [reply]
@tarek133 I just updated to 11.78.1.39 on my unrooted/unmodded device to check, and it still has the classic circle for audio notes, so I guess it's either AB testing like rodit has suggested happens with some features, or rodit actually updated the audio playback in SnapMod. I find it peculiar that pressing and holding on the notes no longer brings up the "Set Playback Speed" option, so that has me thinking that this is an update by rodit, since having the playback speed toggle on the right side of the audio note box means you no longer need the option in the context menu, although like I said, it only goes up to 2x. Not sure what the max speed could be set to with the context menu where you would type the number in manually.

Were you sure to download 11.78.1.39 from APKMirror? It's not available in the Play Store yet, so just updating Snapchat through there won't do you any good. And you of course updated both bindings and APK for SnapMod, right?

**Reply by Magna-Graecia** [reply]
@tarek133 Oh and I forgot to add: Getting an audio note message from someone now says "New Voice Note" instead of just "New Chat". I tried sending an audio note from my SnapMod device to my unrooted/unmodded device running 11.78.1.39, and it still says "New Chat". This really has me wondering if this is a new AB feature by Snapchat, or if it's actually some tweaking on the part of rodit for SnapMod haha.

**Reply by NMBlanco** [reply]
@Magna-Graecia yeah just double checked and I'm on 11.78.1.39, downloaded from apkmirror + latest snapmod bindings and apk. Also just force closed snapchat and checked if I might got that new UI now but nope. All the same still and tbh I'm glad 😅 Really think it's server sided from Snap Inc., if rodit did anything big like that he'd have mentioned it in the changelog surely

**Reply by Magna-Graecia** [reply]
@tarek133 Ya exactly, I was thinking the same thing. Rodit would likely have mentioned some changes pertaining to audio notes. But at the same time I find it funny that Snapchat would have updated audio notes and added a playback speed feature RIGHT after rodit added it through SnapMod lmao. I mean that's the first I've seen of that mod for Snapchat from any of the Snapchat modding tools that I've used over the years. Then within 4 days of the feature coming to Snapchat through mods, suddenly Snapchat adds it themselves? Someone from Snapchat would have to be lurking this repository for sure then in that case haha. Like "hmm, a playback speed feature, these guys are right. That IS a good feature. Why didn't we think of that? Welp, next update." 😂

Plus "Set Playback Speed" is missing from the context menu now. It's a little odd that it's gone right after a playback speed toggle is added to the audio note message box itself. Although I guess an update from Snapchat in this regard could have broken it. We'll just have to see what rodit says about it.

**Reply by rodit** [reply]
This is not my doing haha - again it's certainly the result of some testing - I still have the original voice note view. When I was developing the playback speed feature, there was already code which suggested playback speed was designed to be modifiable, so the sc devs probably had this in mind for some time.

I wouldnt expect the 'Set playback speed' menu option to appear for the new UI as it uses a different view model class to the old one. The same way the 'set playback speed' option doesn't appear for a normal chat message or a snap, it won't appear for the new audio note view either.

**Reply by Magna-Graecia** [reply]
This makes complete sense now. I just want to note that the other menu options pertaining to SnapMod still appear. "Save to Camera Roll" still appears, which normally doesn't for audio notes in unmodded Snapchat, but it does with SnapMod if "Save Any Message" is enabled. And also "Delete Original Message" still shows. This one never worked though. Does it work for you? I'm assuming what it does is it deletes messages sent by the other person? (pretty funny and trollish to be honest lmao. Would love to mess around with it if it worked).

So does this mean you have to update the "Set Playback Speed" option for the new UI once you get it? Did that option have a max limit, or could you set it to whatever number and it would work? I would imagine there has to be SOME limit haha. But I mainly ask because the built-in Snapchat one limits you to 2x, so if it can go above that with Snapmod, then I suppose that's beneficial. The benefit of the Snapchat one is that it's a toggle built right into the audio note message box that you just tap to toggle speeds. No need to press and hold and go through the context menu. Not really a big deal to me personally. Any speed limits also aren't a big deal to me, as I haven't even used this feature yet and don't really see myself doing so. I don't fully understand what the need for it is, I just think it's neat that we can do it for the sake of doing it lmao. As a modder, I'm sure you completely understand the sentiment. Doing something completely useless and meaningless just for the sake of knowing you can do it and seeing if it works hahaha.

**Reply by rodit** [reply]
Please see my reply under the #118 issue about saving audio notes and playback speed modifier. I agree it's nice to have the playback speed modifier on the audio note itself - I might just increase the range of options that appear (rather than just 2x) if it's easy to do.

**Reply by rodit** [reply]
Also no you cannot delete messages other users have sent haha I did try that originally I just haven't bothered to remove that testing code yet which is why the option to do so is still there.
